tgui/packages/tgui/components/Dropdown.tsx

```diff
@@ -234,6 +237,104 @@ export class Dropdown extends Component<DropdownProps, DropdownState> {
     }
   }
 
+  getOptionValue(option) {
+    return typeof option === 'string' ? option : option.value;
+  }
+
+  getOptionsValues() {
+    const { options = [] } = this.props;
+
+    return options.map((option) => {
+      return this.getOptionValue(option);
+    });
+  }
+
+  getSelectedIndex() {
+    const selected = this.state.selected
+      ? this.state.selected
+      : this.props.selected;
+    const { options = [] } = this.props;
+
+    let selectedIndex;
+    options.forEach((option, index) => {
+      let value = this.getOptionValue(option);
+
+      if (value === selected) {
+        selectedIndex = index;
+      }
+    });
+
+    return selectedIndex;
+  }
+
+  hasSwitchToPrevious() {
```

Comment:
hasSwitch and switch has almost 100% the same logic between them
Need to merge them to avoid copypasting